internal/controller/certificates/policies/checks.go

```diff
@@ -94,6 +94,39 @@ func SecretPrivateKeyMatchesSpec(input Input) (string, string, bool) {
 	return "", "", false
 }
 
+// SecretKeystoreFormatMatchesSpec - When the keystore is not defined, the keystore
+// related fields are removed from the secret.
+// When one or more key stores are defined, re-issuance ensure that the
+// corresponding secrets are generated.
+// If the private key rotation is set to "Never", the key store related values are re-encoded
+// as per the certificate specification
+func SecretKeystoreFormatMatchesSpec(input Input) (string, string, bool) {
+	if input.Certificate.Spec.Keystores == nil {
+		if len(input.Secret.Data[cmapi.Pkcs12SecretKey]) != 0 ||
+			len(input.Secret.Data[cmapi.Pkcs12TruststoreKey]) != 0 ||
+			len(input.Secret.Data[cmapi.JksSecretKey]) != 0 ||
+			len(input.Secret.Data[cmapi.JksTruststoreKey]) != 0 {
+			return SecretMismatch, "Keystore is not defined", true
+		}
+		return "", "", false
+	}
+
+	if input.Certificate.Spec.Keystores.JKS != nil {
+		if len(input.Secret.Data[cmapi.JksSecretKey]) == 0 ||
+			len(input.Secret.Data[cmapi.JksTruststoreKey]) == 0 {
+			return SecretMismatch, "JKS Keystore keys does not contain data", true
+		}
+	}
+
+	if input.Certificate.Spec.Keystores.PKCS12 != nil {
+		if len(input.Secret.Data[cmapi.Pkcs12SecretKey]) == 0 ||
+			len(input.Secret.Data[cmapi.Pkcs12TruststoreKey]) == 0 {
+			return SecretMismatch, "PKCS12 Keystore keys does not contain data", true
+		}
+	}
```

Comment:
What about the case where PKCS12 is removed or `create` is set to false? 
Then it should re-issue and remove the corresponding file from the Secret, if it exists.
But it does not.

Reply:
`kubectl patch certificate -n cert-manager-test selfsigned-cert --type merge --patch '{"spec":{"keystores":{"pkcs12": null}}}'`

Reply:
This is fixed